### PR TITLE
Fix create and delete organisation

### PR DESCRIPTION
**Pull request by Sujanadh**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [X] 🐛 Bug Fix
- [ ] 📝 Documentation
- [ ] 🧑‍💻 Refactor
- [ ] ✅ Test
- [ ] 🤖 Build or CI
- [ ] ❓ Other (please specify)

## Related Issue

- Issue #1866 

## Describe this PR

This PR fixes the issue related to creating and deleting organizations awaiting the async functions that were missing. It also fixes the `get_my_organisations` using proper parameter injection and `row_factory` to define the db cursor to use Pydantic model `OrganisationOut` for the response.

***frontend***
There is a payload mismatch with the body and query in frontend (yet to be fixed), @NSUWAL123 

## Screenshots

N/A

## Alternative Approaches Considered

Did you attempt any other approaches that are not documented in code?

## Review Guide

Notes for the reviewer. How to test this change?

## Checklist before requesting a review

- 📖 Read the FMTM Contributing Guide: <https://github.com/hotosm/fmtm/blob/main/CONTRIBUTING.md>
- 📖 Read the HOT Code of Conduct: <https://docs.hotosm.org/code-of-conduct>
- 👷‍♀️ Create small PRs. In most cases, this will be possible.
- ✅ Provide tests for your changes.
- 📝 Use descriptive commit messages.
- 📗 Update any related documentation and include any relevant screenshots.

### CODE REVIEW

**Comment by Sujanadh** [comment]
@spwoodcock , here I haven't updated the status of NO_CONTENT, which would be the best to use? Use it as it is or update it to return a success message?

**Comment by spwoodcock** [comment]
> @spwoodcock , here I haven't updated the status of NO_CONTENT, which would be the best to use? Use it as it is or update it to return a success message?

Good question as it's a bit counter intuitive, but NO_CONTENT is a pretty common response code for delete endpoints (often recommended). So the code looks good as it is 👍

File: src/backend/app/organisations/organisation_deps.py [hunk 1]
```diff
@@ -58,6 +58,8 @@ async def get_organisation(
             pass
         db_org = await DbOrganisation.one(db, id)
 
+        if db_org is None:
+            raise KeyError(f"Organisation ({id}) not found.")
```
[the remark this answers is not in the document]
Comment (Sujanadh) [reply src/backend/app/organisations/organisation_deps.py:62]: Ah I see, I got your point. But the reason why I removed it from the `.one` function is that if you see on the create function of the organisation , there is a logic to check if the organisation already exists or not to allow the creation of a new organisation where `.one` function is called, so if there isn't any organisation with the same name, it should have allowed creating new organisation but the `.one` function raises a `KeyError` as it won't find any db record which blocks `.create` function to create a new organisation.

File: src/backend/app/organisations/organisation_routes.py [hunk 2]
```diff
@@ -120,19 +121,14 @@ async def delete_org(
     org_user_dict: Annotated[AuthUser, Depends(org_admin)],
 ):
     """Delete an organisation."""
-    org = org_user_dict.get("org")
-    deleted_org_id = await DbOrganisation.delete(db, org.id)
-
-    if not deleted_org_id:
-        return Response(
+    org_deleted = await DbOrganisation.delete(db, org_user_dict.id)
```
[the remark this answers is not in the document]
Comment (Sujanadh) [reply src/backend/app/organisations/organisation_routes.py:124]: oh yeah actually, I tested and overdid it while refactoring without noticing. Thanks for pointing out.

File: src/backend/app/organisations/organisation_deps.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (spwoodcock) [reply src/backend/app/organisations/organisation_deps.py:62]: Ah that makes sense 👍

I'm not on my laptop to really dig into the logic. 

But it work using the `ignore_conflict` param on DbOrganisation.create?

We just need to ideally check if the name already exists on creation, but don't need to get the org with `.one`. Not sure what the best solution is, so I'll let you decide what you think is best 😁

**Review by spwoodcock — APPROVED** [review]
Merge if you are happy with it!

File: src/backend/app/organisations/organisation_deps.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sujanadh) [reply src/backend/app/organisations/organisation_deps.py:62]: Yes, we need to check if that name already exists, so we need to write SQL again to check it in the create function. Or, we might create a new function to check if the organisation exists or not and use it in other endpoints too. We already have the `org_exists` function in `organisation_deps` already we can replace it with a new class function in by defining it in DbOrganisation?

**Comment by Sujanadh** [comment]
I have made some changes keeping `KeyError` as it is in ` .create ` function. I also created separate parser function to generate form fields for the organisation input. @spwoodcock

File: src/backend/app/db/models.py [hunk 3]
```diff
@@ -397,11 +397,6 @@ async def create(
         # Set requesting user to the org owner
         org_in.created_by = user_id
 
-        if not ignore_conflict and cls.one(db, org_in.name):
```
Comment (spwoodcock) [inline src/backend/app/db/models.py:400]: Is ignore_conflict used anywhere now?

I forgot why I added it in the first place.
Perhaps it can be removed.

**Review by spwoodcock — APPROVED** [review]
Nice work!

File: src/backend/app/db/models.py (hunk 3 above)
Comment (Sujanadh) [reply src/backend/app/db/models.py:400]: Yes, it is used in the init default HOTOSM organisation.

Comment (Sujanadh) [reply src/backend/app/db/models.py:400]: It is still there in SQL to create an organization.

**Pull request merged**